sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResult.java

```diff
@@ -160,6 +163,7 @@ public String toString() {
                 ", requestCharge: " + this.requestCharge +
                 ", itemLSN: " + this.itemLSN +
                 ", sessionToken: " + (this.sessionToken != null ? this.sessionToken.convertToString() : null) +
+                ", backendLatencyInMs: " + this.backendLatencyInMs +
```

Comment:
Should this match the .NET name? BELatencyInMs

Reply:
Yes, that was my thought too, but java's convention on naming is different than .Net. It doesn't start with upper case. So the other options were `beLatencyInMs` or `bELatencyInMs` and either of them were not super clear. That's why went with `backendLatencyInMs`. That being said, I am open to renaming it.

Reply:
In general we don't need to stick with the names as in the backend implementation. What is "backend" in this case for a regular user reading through that diagnostic string?? I recommend using plain English describing what that value represents, something along "serviceProcessTimeInMs", since that is the real value, and not the actual latency because it is missing the pre-processing and post-processing time the request spent while in queue or elsewhere on the service side.

Reply:
there are many other implementation related info in store-result, lsn, itemLSN. where none of it would make sense to the end user. the diagnostics details is really for the dev team and not the customer. I think we should stick to whatever term .Net/BackEnd used for this for consistency.

Also in case we are hitting gateway does `serviceProcessTimeInMs` mean the gateway processing time? does it mean the backend processing time? for these reasons I prefer we stick to whatever .dotNet named this for consistency.

Reply:
@milismsft  @moderakh  @j82w - .Net named this - `BELatencyInMs` - which I am against because it is not java convention. 
Also, @milismsft  - I also agree with @moderakh  here, `serviceProcessTimeInMs` might not be understandable by customers. 
Currently all our customers know that the word `backend` represents our cosmos db service. That's why both .net and Java SDK are using it. However, .net is using it in short form like `BE` and I am using it in full form here. 
As I said earlier, I am open to renaming it, but I feel whatever we name, it should be clear for both us and customer.

Reply:
Latency and process times are two different things, connected (latency includes process times) but also distinct at the same time. Judging from the server side code implementation, these are processing times, not latency, since they only include the time spent executing the request.
I suggest we present the information when possible in a format easy to understand. And if consistency between .Net and Java is important may be .Net should change as well. Nobody has time these days to go over the documentation (if any) to correctly assess what these metrics represent; making an effort to capture the essence of that metric in its name should be something we should alway leverage when possible to do so (like in this case).

Reply:
+1 for processTime over Latency @milismsft good suggestion. I agree with the reasoning this is really processingTime not Latency.

my comment was on service vs BE. which I think we should keep BE instead of Service as Service is ambiguous not clear if we are referring to GW or BE.

Reply:
BELatencyInMs is just the JSON output so it doesn't need to follow Java or c# convention. I don't see a reason to spell out BackendEnd. 

I would argue that we should use latency because that is the name of the header. If anything the backend implementation should be modified to include the everything rather than just the process time.

Reply:
BE->backend bloating diagnostics is not valid point IMHO, with this logic we should have used acronym for all variable in diagnostics and costing the readability.
On Latency vs ServiceProces, latency is what we are using across diagnostics, and that is what coming from server . Also honestly serviceProcess to me is way to less intuitive than latency 
So I think what we have right now backendLatencyInMs is simple plain english easy to understand

Reply:
@j82w - It does make a difference for naming convention, as Java uses jackson library for our JSON processing, and it has default implementation of camel case naming conventions, which is also used when serializing the objects that we write on diagnostics, which is why all diagnostics in Java SDK are camel cased. https://fasterxml.github.io/jackson-databind/javadoc/2.7/com/fasterxml/jackson/databind/PropertyNamingStrategy.html

Regarding naming, I agree, on using `latency` rather than `processing` time, as latency is used across the diagnostics and it is more clear to the customer as they are already familiar with it.